encord/orm/dataset.py

```diff
@@ -54,41 +57,189 @@ class DatasetUsers:
     pass
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
 class DataClientMetadata:
     payload: dict
 
 
+class ImageData:
+    """
+    Information about individual images within a single :class:`~encord.orm.dataset.DataRow` of type
+    :meth:`DataType.IMG_GROUP <encord.constants.enums.DataType.IMG_GROUP>`. Get this information
+    via the :meth:`DataRow.images <encord.orm.dataset.DataRow.images>` property.
+    """
+
+    def __init__(
+        self,
+        image_hash: UUID,
+        title: str,
+        file_link: str,
+        file_type: str,
+        file_size: int,
+        storage_location: StorageLocation,
+        created_at: datetime,
+        last_edited_at: datetime,
+        width: int,
+        signed_url: Optional[str],
+        height: int,
+    ):
+
+        self._image_hash = image_hash
+        self._title = title
+        self._file_link = file_link
+        self._file_type = file_type
+        self._file_size = file_size
+        self._storage_location = storage_location
+        self._created_at = created_at
+        self._last_edited_at = last_edited_at
+        self._height = height
+        self._width = width
+        self._signed_url = signed_url
+
+    @property
+    def image_hash(self) -> str:
+        return str(self._image_hash)
+
+    @property
+    def title(self) -> str:
+        return self._title
+
+    @property
+    def file_link(self) -> str:
+        return self._file_link
+
+    @property
+    def file_type(self) -> str:
+        """
+        The MIME type of the file.
+        """
+        return self._file_type
+
+    @property
+    def file_size(self) -> int:
+        """
+        The size of the file in bytes.
+        """
+        return self._file_size
+
+    @property
+    def storage_location(self) -> StorageLocation:
+        return self._storage_location
+
+    @property
+    def created_at(self) -> datetime:
+        return self._created_at
+
+    @property
+    def last_edited_at(self) -> datetime:
+        return self._last_edited_at
+
+    @property
+    def height(self) -> int:
+        return self._height
+
+    @property
+    def width(self) -> int:
+        return self._width
+
+    @property
+    def signed_url(self) -> Optional[str]:
+        """The signed URL if one was generated when this class was created."""
+        return self._signed_url
+
+    @classmethod
+    def from_dict(cls, json_dict: Dict) -> ImageData:
+        return ImageData(
+            title=json_dict["title"],
+            file_link=json_dict["file_link"],
+            file_type=json_dict["file_type"],
+            file_size=json_dict["file_size"],
+            image_hash=json_dict["image_hash"],
+            storage_location=json_dict["storage_location"],
```

Comment:
`storage_location` is an `IntEnum`, should we instantiate it?

```suggestion
            storage_location=StorageLocation(json_dict["storage_location"]),
```

Reply:
yay